apps/mobile/src/features/swap/swap-state/use-swap-state.ts

```diff
@@ -1,43 +1,72 @@
-import { useMemo, useReducer } from 'react';
+import { useReducer } from 'react';
 
-import { useDerivedAmounts } from '@/features/swap/swap-state/hooks/use-derived-amounts';
-import { useIsSendingMax } from '@/features/swap/swap-state/hooks/use-is-sending-max';
-import { useSecondaryAmount } from '@/features/swap/swap-state/hooks/use-secondary-amount';
-import { useSwapExecutability } from '@/features/swap/swap-state/hooks/use-swap-executability';
-import { DEFAULT_SLIPPAGE_PERCENTAGE } from '@/features/swap/swap-state/swap.constants';
-import {
-  useAccountBaseSwapAssetsQuery,
-  useAccountTargetSwapAssetsQuery,
-  useAssetMarketDataQuery,
-} from '@/features/swap/swap-state/swap.queries';
+import { useNetworkFee } from '@/features/swap/swap-state/hooks/use-network-fee';
+import * as btc from '@scure/btc-signer';
+import { StacksNetwork } from '@stacks/network';
 
-import { AccountAddresses, QuoteCurrency, SwappableFungibleCryptoAsset } from '@leather.io/models';
-import { AccountSwapAsset, MarketDataService, SwapService } from '@leather.io/services';
+import { BitcoinNativeSegwitPayer } from '@leather.io/bitcoin';
+import {
+  AccountAddresses,
+  NetworkConfiguration,
+  QuoteCurrency,
+  SwappableFungibleCryptoAsset,
+} from '@leather.io/models';
+import {
+  AccountSwapAsset,
+  BitcoinTransactionFeesService,
+  MarketDataService,
+  StacksTransactionFeesService,
+  SwapService,
+} from '@leather.io/services';
+import { StacksSigner } from '@leather.io/stacks';
 import { getAssetId } from '@leather.io/utils';
 
 import { createSwapActions } from './actions/swap-actions';
+import { useDerivedAmounts } from './hooks/use-derived-amounts';
+import { useIsSendingMax } from './hooks/use-is-sending-max';
+import { useSecondaryAmount } from './hooks/use-secondary-amount';
 import { useSwapAssetReconciliation } from './hooks/use-swap-asset-reconciliation';
+import { useSwapExecutability } from './hooks/use-swap-executability';
 import { useSwapQuotes } from './hooks/use-swap-quotes';
 import { useSwapValidation } from './hooks/use-swap-validation';
 import { swapReducer } from './swap-state.reducer';
 import { SwapInternalState, UseSwapStateResult } from './swap-state.types';
+import { DEFAULT_SLIPPAGE_PERCENTAGE } from './swap.constants';
+import {
+  useAccountBaseSwapAssetsQuery,
+  useAccountTargetSwapAssetsQuery,
+  useAssetMarketDataQuery,
+} from './swap.queries';
 
 export interface UseSwapStateProps {
   accountRequest: { account: AccountAddresses };
   baseAsset?: SwappableFungibleCryptoAsset;
   targetAsset?: SwappableFungibleCryptoAsset;
-  marketDataService: MarketDataService;
-  swapService: SwapService;
   quoteCurrencyPreference: QuoteCurrency;
+  swapService: SwapService;
```

Comment:
@kyranjamie Per our past conversation, the core dependencies for this machine have grown quite a bit. I'm working on a standardized provider to be used in the extension and mobile as a better injection mechanism, will add it once the signing/broadcasting logic is in.

The current direct passing isn't yet terrible, but it will definitely get out of hand as the feature grows.